peer_mentoring/groups/views.py

```diff
@@ -1,7 +1,61 @@
-from django.shortcuts import render
+from account_management.models import UserProfile
+from django.contrib.auth.decorators import login_required
+from django.shortcuts import get_object_or_404, redirect, render
+from posts.models import Comment, Post
+
+from .forms import GroupPostCommentForm, GroupPostForm
+from .models import Group
 
 # Create your views here.
 
 
-def group_view(request):
-    pass
+@login_required
+def group_index_view(request):
+    groups = Group.objects.all()
+    context = {"groups": groups}
+    return render(request, "groups_index.html", context)
+
+
+@login_required
+def group_detail_view(request):
+    group = get_object_or_404(Group, pk=Group.id)
+    context = {
+        "group": group,
+    }
+    return render(request, "group_detail.html", context)
+
+
+@login_required
+def group_posts(request):
+    posts = Post.objects.all().filter(id=Group.id)
+    comments = request.POST.get(Comment)
+    return render(request, "group_detail.html", posts, comments)
+
+
+@login_required
+def group_members_index_view(request):
+    members = UserProfile.objects.all()
+    context = {"members": members}
+    return render(request, "group_detail.html", context)
+
+
+@login_required
+def group_create_post_view(request):
+    if request.method == "POST":
+        form = GroupPostForm(request.POST)
+        post = form.save()
+        return redirect("group-detail", post.id)
```

Comment:
You can use the messages framework later to show a success message

Reply:
Okay, should this become an issue then or should I resolve now?

Reply:
Added the message with the redirect, that's what I was thinking. It could post the success message in addition to or as a replacement for the redirect. What do you think? Also, am I correct in the logic?